VideoApi/VideoApi/Services/VirtualRoomService.cs

```diff
@@ -0,0 +1,103 @@
+        }
+
+        private async Task<Room> CreateAnInterpreterRoom(Conference conference, int existingRooms)
+        {
+            _logger.LogInformation("Creating a new interpreter room for conference {Conference}", conference.Id);
+            var createRoomCommand = new CreateRoomCommand(conference.Id, null, VirtualCourtRoomType.Civilian, false);
+            await _commandHandler.Handle(createRoomCommand);
+            var roomId = createRoomCommand.NewRoomId;
+            _logger.LogInformation("Creating a new civilian room {Room} for conference {Conference}", roomId,
+                conference.Id);
+            var ingestUrl = $"{conference.IngestUrl}/{roomId}";
+            var roomPrefix = "Interpreter";
+
+            var newRoomParams = new CreateParticipantRoomParams
+            {
+                Audio_recording_url = ingestUrl,
+                Participant_room_id = roomId.ToString(),
+                Participant_type = VirtualCourtRoomType.Civilian.ToString(),
+                Room_label_prefix = roomPrefix,
+                Room_type = roomPrefix,
+                Display_name = $"{roomPrefix}{existingRooms + 1}",
+                Tile_number = $"T{existingRooms + 1}"
+            };
+            var newVmr = await _kinlyApiClient.CreateParticipantRoomAsync(conference.Id.ToString(), newRoomParams);
+
+            var updateCommand = new UpdateRoomCommand(conference.Id, roomId, newVmr.Display_name, ingestUrl,
+                newVmr.Uris.Pexip_node, newVmr.Uris.Participant);
+            await _commandHandler.Handle(updateCommand);
+            _logger.LogDebug("Updated room {Room} for conference {Conference} with joining details", roomId,
+                conference.Id);
+            var query = new GetAvailableRoomByRoomTypeQuery(VirtualCourtRoomType.Civilian, conference.Id);
+            var listOfRooms = await _queryHandler.Handle<GetAvailableRoomByRoomTypeQuery, List<Room>>(query);
+
+            return listOfRooms.First(x => x.Id == roomId);
+        }
+
+        private Room GetRoomForParticipant(Guid conferenceId, IReadOnlyCollection<Room> rooms, Participant participant)
+        {
+            _logger.LogInformation(
+                "Checking for an existing interpreter room for participant {Participant} in conference {Conference}",
+                participant.Id, conferenceId);
+            var participantIds = participant.LinkedParticipants.Select(lp => lp.LinkedId).ToList();
+            participantIds.Add(participant.Id);
+
+            // first check if a room with linked participants for current participant exist
+            var matchingRooms = participantIds
+                .Select(l => rooms.SingleOrDefault(r => r.DoesParticipantExist(new RoomParticipant(l))))
+                .Where(m => m != null)
+                .Select(matching => matching)
+                .ToList();
+
+            var existingRoom = matchingRooms.FirstOrDefault();
+            // return an empty room, if possible
```

Comment:
remove comments, can refactor if you feel it needs to be clearer (but i dont think it does :) )

Reply:
will do. the method names are pretty clear.